frontend/packages/dev-console/src/actions/add-resources.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import {
+  GitAltIcon,
+  OsImageIcon,
+  CatalogIcon,
+  CubeIcon,
+  DatabaseIcon,
+  UnknownIcon,
+} from '@patternfly/react-icons';
+import { K8sResourceKind, modelFor, referenceFor } from '@console/internal/module/k8s';
+import { KebabOption, asAccessReview } from '@console/internal/components/utils';
+import { ImportOptions } from '../components/import/import-types';
+import { getMenuPath, getAddPageUrl } from '../utils/add-resources-menu-utils';
+
+export const FromGit = (obj?: K8sResourceKind, hasApplication?: boolean): KebabOption => {
+  const resourceModel = modelFor(referenceFor(obj));
+  return {
+    path: getMenuPath(hasApplication),
+    label: 'From Git',
+    href: getAddPageUrl(obj, ImportOptions.GIT, hasApplication),
+    icon: <GitAltIcon />,
+    accessReview: asAccessReview(resourceModel, obj, 'create'),
+  };
+};
+
+export const ContainerImage = (obj: K8sResourceKind, hasApplication?: boolean): KebabOption => {
```

Comment:
You're repeating yourself quite a bit here... I think something like _currying_ (lots of resources on the web for this concept if you want to know more) could be of use here where you store some static variables at the start and return a function that can make the actual `KebabOption` at a later time (when invoked).

```
// From my other comment
type KebabAction = (obj?: K8sResourceKind, hasApplication?: boolean) => KebabOption;

// Currying-based type
type KebabFactory = (
  label: string,
  icon: React.ReactNode,
  importType: ImportOptions,
) => KebabAction;

// One setup that consumes both stages
export const createKebabOption: KebabFactory = (label, icon, importType) => (
  obj,
  hasApplication,
) => {
  const resourceModel = modelFor(referenceFor(obj));
  return {
    path: getMenuPath(hasApplication),
    label,
    href: getAddPageUrl(obj, importType, hasApplication),
    icon,
    accessReview: asAccessReview(resourceModel, obj, 'create'),
  };
};

// Setup each option with statics values that returns a function to be
// consumed later and provide the rest of the options
export const fromGit = createKebabOption('From Git', <GitAltIcon />, ImportOptions.GIT);
export const containerImage = createKebabOption(
  'Container Image',
  <OsImageIcon />,
  ImportOptions.CONTAINER,
);
```
I didn't spend any time on the variable / function / type names, so by all means adjust as you see fit. But I think this will help you repeat less in this file and give easy access to adding new flows.

Reply:
refactored it, definitely looks better now.